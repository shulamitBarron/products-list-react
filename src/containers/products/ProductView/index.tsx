import { Product } from 'actions/redux/product/interfaces';
import * as React from 'react';
import { Card, Button } from 'react-bootstrap';
import { TranslateFunction } from 'react-localize-redux/es';
import { Link } from 'react-router-dom';

interface Props {
	product: Product;
	translate: TranslateFunction;
	deleteProduct: (id: string) => void;
}

const ProductView: React.FC<Props> = (props: Props) => {
	const { product, translate, deleteProduct } = props;

	const handleDelete = () => {
		deleteProduct(product.id);
	};

	return (
		<Card>
			<Card.Header>{product.name}</Card.Header>
			<Card.Body>
				{product.description}
				<br />
				<Link to={{ pathname: `product/${product.id}` }}> {translate('products.updateProduct')}</Link>
				<br />
				<Button onClick={handleDelete}>{translate('products.deleteProduct')}</Button>
			</Card.Body>
			<Card.Img src={product.picture} />
		</Card>
	);
};

export default ProductView;
