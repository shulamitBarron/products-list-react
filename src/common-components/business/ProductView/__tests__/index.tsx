import React from 'react';
import { shallow } from 'enzyme';
import { Card, Button } from 'react-bootstrap';
import ProductView from '../index';
import { Link } from 'react-router-dom';

const product = {
	id: '5ff81abb0e750ff69443daf8',
	isInStock: true,
	price: '3554.61',
	picture: 'http://placehold.it/32x32',
	name: 'Hewitt',
	group: 'sport',
	description: 'Officia adipisicing velit anim deserunt officia est qui excepteur ullamco laborum.',
	registered: '2018-12-14 15:32:56'
};

describe('ProductView', () => {
	it('should render component', () => {
		const tree = shallow(<ProductView product={product} deleteProduct={jest.fn()} translate={jest.fn()} isEditble />);
		expect(tree).toMatchSnapshot();
	});

	it('should render not editable component', () => {
		const tree = shallow(<ProductView product={product} translate={jest.fn()} isEditble={false} />);
		expect(tree).toMatchSnapshot();
	});

	it('should render with correct childern', () => {
		const tree = shallow(<ProductView product={product} deleteProduct={jest.fn()} translate={jest.fn()} isEditble />);

		expect(tree.find(Card.Header).text()).toBe('Hewitt');
	});

	it('should render not editable component with correct childern', () => {
		const tree = shallow(<ProductView product={product} translate={jest.fn()} isEditble={false} />);

		expect(tree.find(Link).exists()).toBe(false);
		expect(tree.find(Button).exists()).toBe(false);
	});

	it('should call deleteProduct', () => {
		const deleteProduct = jest.fn();
		const tree = shallow(<ProductView product={product} deleteProduct={deleteProduct} translate={jest.fn()} isEditble />);

		tree.find(Button).simulate('click');
		expect(deleteProduct).toHaveBeenCalledWith('5ff81abb0e750ff69443daf8');
	});
});
