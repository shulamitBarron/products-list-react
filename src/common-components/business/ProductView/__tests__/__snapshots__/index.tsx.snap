// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductView should render component 1`] = `
<Card
  body={false}
>
  <CardHeader>
    Hewitt
  </CardHeader>
  <CardBody>
    Officia adipisicing velit anim deserunt officia est qui excepteur ullamco laborum.
    <br />
    <Link
      to={
        Object {
          "pathname": "product/5ff81abb0e750ff69443daf8",
        }
      }
    >
       
    </Link>
    <br />
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    />
  </CardBody>
  <CardImg
    src="http://placehold.it/32x32"
    variant={null}
  />
</Card>
`;

exports[`ProductView should render not editable component 1`] = `
<Card
  body={false}
>
  <CardHeader>
    Hewitt
  </CardHeader>
  <CardBody>
    Officia adipisicing velit anim deserunt officia est qui excepteur ullamco laborum.
  </CardBody>
  <CardImg
    src="http://placehold.it/32x32"
    variant={null}
  />
</Card>
`;
